packages/grafana-data/src/panel/PanelPlugin.ts

```diff
@@ -259,6 +265,34 @@ export class PanelPlugin<
     return this;
   }
 
+  /**
+   * Tells Grafana if the plugin should subscribe to annotation and alertState results.
+   *
+   * @example
+   * ```typescript
+   *
+   * import { ShapePanel } from './ShapePanel';
+   *
+   * interface ShapePanelOptions {}
+   *
+   * export const plugin = new PanelPlugin<ShapePanelOptions>(ShapePanel)
+   *     .useFieldConfig({})
+   *     ...
+   *     ...
+   *     .setAnnotationSupport({
+   *       annotations: true,
+   *       alertStates: true,
+   *     });
+   * ```
+   *
+   * @public
+   **/
+  setAnnotationSupport({ annotations, alertStates }: { annotations?: boolean; alertStates?: boolean }) {
+    this.supportsAlertStates = alertStates ?? this.supportsAlertStates;
+    this.supportsAnnotations = annotations ?? this.supportsAnnotations;
+    return this;
+  }
```

Comment:
ok, sorry to take this one more round :) 

how about 

```
setDataSupport(dataSupport: PanelPluginDataSupport) {
  this.dataSupport = dataSupport; 
}
```

PanelPluginDataSupport  would have annotations, alertStates,  and possibly other flags in the future like "singleDataFrameOnly" or something like that.